components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -0,0 +1,2222 @@
+                    "SELECT details.group.id FROM OriginalFile o WHERE o.id = :id",
+                    new ParametersI().addId(remoteFile.getId())).get(0).get(0)).getValue();
+            final long imageGroupId = ((RLong) iQuery.projection(
+                    "SELECT details.group.id FROM Image i WHERE i.id = :id",
+                    new ParametersI().addId(image.getId())).get(0).get(0)).getValue();
+            final long datasetGroupId = ((RLong) iQuery.projection(
+                    "SELECT details.group.id FROM Dataset d WHERE d.id = :id",
+                    new ParametersI().addId(sentDat.getId())).get(0).get(0)).getValue();
+            final List<RType> resultForLink = iQuery.projection(
+                    "SELECT id, details.group.id FROM DatasetImageLink WHERE parent.id  = :id",
+                    new ParametersI().addId(sentDat.getId())).get(0);
+            final long linkId = ((RLong) resultForLink.get(0)).getValue();
+            final long linkGroupId = ((RLong) resultForLink.get(1)).getValue();
+            /* check that the image, dataset and link still belongs
+             * to the light admin as the chown failed, but are in the group of normalUser */
+            assertOwnedBy(image, lightAdmin);
+            Assert.assertEquals(imageGroupId, normalUser.groupId);
+            assertOwnedBy(sentDat, lightAdmin);
+            Assert.assertEquals(datasetGroupId, normalUser.groupId);
+            assertOwnedBy((new DatasetImageLinkI(linkId, false)), lightAdmin);
+            Assert.assertEquals(linkGroupId, normalUser.groupId);
+        }
+    }
+
+    /** Additonal test of light amdin without using Sudo.
+     * The workflow deals with the eventuality of pre-existing container
+     * in the target group and linking of the image or dataset to this container
+     * (dataset or project). The image import has been tested in other tests,
+     * here the image and/or dataset will be created and saved instead and just
+     * the linking to a container will be tested. Only when the light admin has
+     * WriteOwned privilege is the linking possible.
+     * @param isAdmin if to test a member of the <tt>system</tt> group
+     * @param permChown if to test a user who has the <tt>Chown</tt> privilege
+     * @param permWriteOwned if to test a user who has the <tt>WriteOwned</tt> privilege
+     * @param groupPermissions if to test the effect of group permission level
+     * @throws Exception unexpected
+     */
+    @Test(dataProvider = "WriteOwned and Chown privileges cases")
+    public void testLinkNoSudo(boolean permWriteOwned, boolean permChown,
+            String groupPermissions) throws Exception {
+        /* linking should be always permitted as long as light admin is in System Group
+         * and has WriteOwned permissions. Exception is Private group, where linking will
+         * always fail.*/
+        boolean isExpectLinkingSuccess = permWriteOwned && !(groupPermissions == "rw----");
```

Comment:
https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R959
https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R1676
https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R1954

are the fixes for this, which are on a latter PR.